app/recipe/views.py

```diff
@@ -0,0 +1,60 @@
+from rest_framework import viewsets, mixins
+
+from core.models import Recipe, Ingredient
+from recipe import serializers
+from drf_spectacular.utils import (
+    extend_schema,
+    extend_schema_view,
+    OpenApiParameter,
+    OpenApiTypes,
+)
+
+
+@extend_schema_view(
+    list=extend_schema(
+        parameters=[
+            OpenApiParameter(
+                "search",
+                OpenApiTypes.STR,
+                description="Fuzzy search of the recipe title",
+            ),
+        ]
+    )
+)
+class RecipeViewSet(viewsets.ModelViewSet):
+    """Manage recipes in the database"""
+
+    queryset = Recipe.objects.all()
```

Comment:
why are we creating here the queryset if it is only being used within the `get_queryset` method?

Reply:
If I don't set the queryset on the ViewSet object, django can't infer the `basename` when registering the route. Is this the correct way to do it? Happy to do it this way and change the `get_queryset` override to be:
```

    def get_queryset(self):
        """Return objects for authenticated user"""
        search = self.request.query_params.get("search")
        queryset = Recipe.objects.all()
        if search:
            queryset = queryset.filter(title__contains=search)

        return queryset.order_by("-id").distinct()
```

https://www.django-rest-framework.org/api-guide/routers/
https://www.django-rest-framework.org/api-guide/generic-views/#genericapiview